### PR TITLE
Restart Ingestion Job on code version update

### PR DESCRIPTION
**Pull request by pyalex**
**What this PR does / why we need it**:
1. Each Job can have individual labels that are being merged with common labels (see `feast.jobs.runners.options.labels`) before job created. Job labels are being generated & managed by `JobCoordinator`. Currently labels consists of `feast.job.coordinator.jobSelector` and feast version. Job labels overwrite common labels for the same keys.
 
2. Added label `feast_version` that keeps version of feast code attached to the job. As soon as new version is being released - JobCoordinator automatically restarts jobs with older version in this label.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: core/src/main/java/feast/core/service/JobCoordinatorService.java [hunk 1]
```diff
@@ -81,6 +85,14 @@ public JobCoordinatorService(
...
     this.specPublisher = specPublisher;
     this.groupingStrategy = groupingStrategy;
+    this.featureSetSubscriptions =
+        feastProperties.getJobs().getCoordinator().getFeatureSetSelector().stream()
+            .map(JobProperties.CoordinatorProperties.Selector::toSubscription)
+            .collect(Collectors.toList());
+    this.blacklistedStores = feastProperties.getJobs().getCoordinator().getBlacklistedStores();
+    this.currentVersion = feastProperties.getVersion();
+    this.jobLabels = new HashMap<>(feastProperties.getJobs().getCoordinator().getJobSelector());
+    this.jobLabels.put("version", this.currentVersion);
```
Comment (woop) [inline core/src/main/java/feast/core/service/JobCoordinatorService.java:95]: should `this.currentVersion` be a getter?

Comment (pyalex) [reply core/src/main/java/feast/core/service/JobCoordinatorService.java]: this is private final property, not sure how getter could be useful?

**Pull request merged**